### PR TITLE
Add dev.overlayed.Overlayed

### PR DESCRIPTION
**Pull request by Hacksore**
> [!WARNING]
> I don't use linux so be please patient with this PR

### Overview
Overlayed is companion app for Discord that lets you have a movable overlay to see who is in the voice chat.

### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  
- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [ ] I have [built][build] and tested the submission locally ([WIP see README.md](https://github.com/Hacksore/flathub/blob/overlayed/README.md)).
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**
  - [x] https://github.com/overlayeddev/overlayed/pull/172

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by Hacksore** [comment]
I've borrowed a lot of the code from https://github.com/flathub/flathub/pull/4950

I will figure out how to test this but I'm assuming there is no way on macOS and I'll have to perhaps try via WSL+Linux.

Any pointers would be greatly appreciated!

File: com.overlayed.Overlayed.metainfo.xml
Comment (bbhtt) [inline com.overlayed.Overlayed.metainfo.xml:1]: This should be in the upstream repo.

File: com.overlayed.Overlayed.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <summary>A discord overlay</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-only</project_license>
+  <url type="homepage">https://overlayed.dev</url>
+  <url type="bugtracker">https://github.com/overlayeddev/overlayed/issues</url>
+  <url type="vcs-browser">https://github.com/overlayeddev/overlayed</url>
+  <url type="contribute">https://github.com/overlayeddev/overlayed</url>
+  <description>
+    <p>Modern, open-source, and free voice chat overlay for Discord</p>
+  </description>
```
Comment (bbhtt) [inline com.overlayed.Overlayed.metainfo.xml:16]: Please add some more description and features. This is too short.

File: com.overlayed.dev.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+id: com.overlayed.Overlayed
+
+runtime: org.gnome.Platform
+runtime-version: "45"
+sdk: org.gnome.Sdk//45
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:5]:
```suggestion
sdk: org.gnome.Sdk
```

File: com.overlayed.dev.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+rename-icon: overlayed
+command: overlayed
+finish-args:
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create # allows for discord rpc
+  - --device=all
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=x11
+  - --socket=wayland
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:16]: This is either or not both. If it doesn't work on wayland use `--socket=x11` otherwise use `--socket=wayland, --socket=fallback-x11`.

File: com.overlayed.dev.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+modules:
+  - name: webkit2gtk-4.0
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:19]: Why is this needed? This has been deprecated for 2+ years.

File: com.overlayed.dev.yaml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        make-install-args:
+          - prefix=${FLATPAK_DEST}
+        sources:
+          - type: archive
+            url: https://dotat.at/prog/unifdef/unifdef-2.12.tar.xz
+            sha256: 43ce0f02ecdcdc723b2475575563ddb192e988c886d368260bc0a63aee3ac400
+        cleanup:
+          - "*"
+
+  - name: openssl11
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:52]: Openssl 1.1 is no longer supported. Why is this needed?

File: com.overlayed.dev.yaml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    build-commands:
+      - ./config --prefix=${FLATPAK_DEST}
+      - make -j $FLATPAK_BUILDER_N_JOBS
+      - make install_sw
+    cleanup:
+      - /bin
+      - /include
+      - /lib/pkgconfig
+      - "*.a"
+      - "*.la"
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:63]: Add a toplevel cleanup for these 4. No need to repeat it for each one

File: com.overlayed.dev.yaml [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      - install -Dm755 squashfs-root/usr/bin/overlayed /app/bin/overlayed-app
+      - install -Dm644 squashfs-root/usr/share/applications/overlayed.desktop /app/share/applications/com.overlayed.Overlayed.desktop
+      - install -Dm644 com.overlayed.Overlayed.metainfo.xml /app/share/metainfo/com.overlayed.OverlayedApp.metainfo.xml
+
+    sources:
+      - type: file
+        path: com.overlayed.Overlayed.metainfo.xml
+
+      - type: file
+        url: https://github.com/overlayeddev/overlayed/releases/download/v0.5.0/overlayed_0.5.0_amd64.AppImage
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:126]: Why not build from source?

File: com.overlayed.dev.yaml [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        dest-filename: Overlayed.AppImage
+        sha256: bbcc773e89ba6cdbf4b3e91156b2702b2a82a91b5df7a8bbd2e274d5104a9725
+        only-arches: [x86_64]
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:129]: Also is aarch64 available?

File: com.overlayed.dev.yaml [hunk 9]
```diff
@@ -0,0 +1,129 @@
+id: com.overlayed.Overlayed
```
Comment (bbhtt) [inline com.overlayed.dev.yaml:1]: There is nothing in https://overlayed.com/ ? Did you read this link?

image

Comment (bbhtt) [reply com.overlayed.dev.yaml:1]: Marking this as blocked until this is resolved.

**Comment by Hacksore** [comment]
@bbhtt thanks for all the drive by comments, I’ll digest them at my own pace!

File: com.overlayed.dev.yaml (hunk 3 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: ¯\_(ツ)_/¯ no idea as the PR body describes I don't use linux and most of the code was copied from #4950.

File: com.overlayed.dev.yaml (hunk 4 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: ¯_(ツ)_/¯ no idea as the PR body describes I don't use linux and most of the code was copied from https://github.com/flathub/flathub/pull/4950.

File: com.overlayed.dev.yaml (hunk 8 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: no

File: com.overlayed.dev.yaml (hunk 9 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: This was a mistake in the copypasta i've corrected it now, keep in mind this PR is STILL a draft!

File: com.overlayed.dev.yaml (hunk 5 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: I might not be required but let's at least get it working first then we can trim the fat.

File: com.overlayed.Overlayed.metainfo.xml
Comment (Hacksore) [reply com.overlayed.Overlayed.metainfo.xml:1]: i dont get that i see it's listed in the flathub package here for Modrinth and no in the Modrinth app repo.

https://github.com/flathub/com.modrinth.ModrinthApp

**Comment by Hacksore** [comment]
Update on this, I've done my best to address things that made sense to me but @bbhtt has a lot more domain knowledge.

@zerodouglas is a collaborator who may or may not help here if they have time.

I've also extended collaborator to @bbhtt as well, feel free to push anything you find offensive. I'm still trying to figure out how to get `flatpak-builder` working on WSL so I can test this 🤞.

File: com.overlayed.Overlayed.metainfo.xml
Comment (bbhtt) [reply com.overlayed.Overlayed.metainfo.xml:1]: Please don't mark comments as resolved until they actually are.

You missed the part where they opened a PR for them to be included upstream:

image

File: com.overlayed.dev.yaml (hunk 3 above)
Comment (bbhtt) [reply com.overlayed.dev.yaml]: Again do not mark comments as resolved until they are.

File: com.overlayed.dev.yaml (hunk 6 above)
Comment (bbhtt) [reply com.overlayed.dev.yaml]: This is obviously not done yet you marked it as resolved.

**Comment by bbhtt** [comment]
If you are going to mark review comments as resolved with responses like this:

image


then there is no point in continuing the review...

> I've also extended collaborator to @bbhtt as well, feel free to push anything you find offensive. 

Why are you inviting me? I'm here to review, it's your submission.

**Comment by Hacksore** [comment]
@bbhtt Sorry for my tone and ignorance but this is a draft PR (not ready for formal review) and I'm not a linux user.

- Raised https://github.com/overlayeddev/overlayed/pull/172 to address the upstream requirements 
- Will not mark things until they are resolved going forward

May I please ask you make effort to explain things or link to references as I have no deep linux knowledge.

Latest build is passing as well but unsure if that means it's working 😅.
https://github.com/overlayeddev/overlayed/actions/runs/10042644667/job/27753549403?pr=172

**Comment by bbhtt** [comment]
What points do you need help/explanation on?

The build has to work here, I/we can start a build after the comments are resolved.

File: com.overlayed.Overlayed.metainfo.xml
Comment (Hacksore) [reply com.overlayed.Overlayed.metainfo.xml:1]: This has been moved to the upstream via https://github.com/overlayeddev/overlayed/pull/172

File: com.overlayed.dev.yaml (hunk 4 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: This has been removed

File: com.overlayed.dev.yaml (hunk 5 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: This has been removed

File: com.overlayed.dev.yaml (hunk 7 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: If you can point me to any other tauri app building from source I can look into it.

File: com.overlayed.Overlayed.metainfo.xml
Comment (bbhtt) [reply com.overlayed.Overlayed.metainfo.xml:1]: Thanks, you should now use it from there. You can use a `url` for it

```
      - type: file
        url: https://raw.githubusercontent.com/...
        sha256: <sha256 checksum of file>
```

File: com.overlayed.dev.yaml (hunk 4 above)
Comment (bbhtt) [reply com.overlayed.dev.yaml]: Do you need these ? If you need them then it is ok to keep it. I just want to know the reason.

File: com.overlayed.Overlayed.metainfo.xml
Comment (Hacksore) [reply com.overlayed.Overlayed.metainfo.xml:1]: Resolved via https://github.com/flathub/flathub/pull/5431/commits/a3536d86dd72763d01ba91beaff3417640df0a7b

File: dev.overlayed.Overlayed.yaml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+      - mkdir -p /app/{bin,share}
+      - cp -r squashfs-root/usr/share/icons /app/share/icons
+      - install -Dm755 squashfs-root/usr/bin/overlayed /app/bin/overlayed
+      - install -Dm644 squashfs-root/usr/share/applications/overlayed.desktop /app/share/applications/dev.overlayed.Overlayed.desktop
+      - install -Dm644 dev.overlayed.Overlayed.metainfo.xml /app/share/metainfo/dev.overlayed.OverlayedApp.metainfo.xml
+
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/overlayeddev/overlayed/main/distribution/dev.overlayed.Overlayed.metainfo.xml
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:44]:
```suggestion
        url: https://raw.githubusercontent.com/overlayeddev/overlayed/14a6d2c8b022f7d44de4a57350e8f30ff35ef7dc/distribution/dev.overlayed.Overlayed.metainfo.xml
```

Don't use a branch here otherwise the checksum will change whenever you update causing the build to fail.

File: com.overlayed.dev.yaml (hunk 7 above)
Comment (bbhtt) [reply com.overlayed.dev.yaml]: > If you can point me to any other tauri app building from source I can look into it.

What version of tauri are you using? If it is not v2 then that explains including webkit2gtk



https://github.com/flathub/in.cinny.Cinny/blob/master/in.cinny.Cinny.yml is a tauri app that builds from source.

File: com.overlayed.dev.yaml (hunk 4 above)
Comment (Hacksore) [reply com.overlayed.dev.yaml]: I don't know if it's a required tuari native dep but since the upstream CI passed and it runs it without crash seems not needed?

File: com.overlayed.dev.yaml (hunk 7 above)
Comment (bbhtt) [reply com.overlayed.dev.yaml]: openssl1 is likely not needed. You can remove that.

**Comment by hfiguiere** [comment]
> I don't use linux so be please patient with this PR

So how do you intend to test this? Submitting an app is a responsibility. If you don't intend to test it, the best is to find a contributor that can to handle this submission.

BTW @bbhtt isn't drive-by reviewing. He is part of the volunteer trusted reviewer/maintainer team.

File: dev.overlayed.Overlayed.yaml [hunk 11]
```diff
@@ -0,0 +1,51 @@
+id: dev.overlayed.Overlayed
+
+runtime: org.gnome.Platform
+runtime-version: "45"
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:4]: this should be runtime 46. Runtim 45 will be EOL in a couple of month.

File: dev.overlayed.Overlayed.yaml [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+rename-icon: overlayed
+command: overlayed
+finish-args:
+  # NOTE: not sure if this is for both IPC and Websockets transports
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create # allows for discord rpc
+  - --device=all
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:16]: as mentionned in the previous review comments. it is either x11 alone, or wayland + fallback-x11. Unless there is a specific reason which should be explained.

"I don't know" isn't a proper answer.

File: dev.overlayed.Overlayed.yaml (hunk 11 above)
Comment (Hacksore) [reply dev.overlayed.Overlayed.yaml]: Updated

File: dev.overlayed.Overlayed.yaml [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+runtime: org.gnome.Platform
+runtime-version: "46"
+sdk: org.gnome.Sdk
+
+rename-icon: Overlayed
+command: overlayed
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=x11
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:14]: You don't need all 3. The previous comment still applies

File: dev.overlayed.Overlayed.yaml [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/overlayeddev/overlayed/14a6d2c8b022f7d44de4a57350e8f30ff35ef7dc/distribution/dev.overlayed.Overlayed.metainfo.xml
+        sha256: 35773ae27a47753852aac2ace795675cef222d5d9681e5cdd93d92b6d16831c5
+
+      # NOTE: skipping building from source as there is no direct pnpm support as of yet
+      # https://github.com/flathub/flathub/pull/3876
+      # https://github.com/flatpak/flatpak-builder-tools/issues/383
+      - type: file
+        url: https://github.com/overlayeddev/overlayed/releases/download/v0.5.0/overlayed_0.5.0_amd64.AppImage
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:92]: you mean the appimage doesn't come with webkitgtk? What the point of it if it can't work on many system?

File: dev.overlayed.Overlayed.yaml [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+      - |
+        desktop-file-edit \
+          --set-name="Overlayed" \
+          --set-comment="Overlayed's desktop application" \
+          --set-icon="$FLATPAK_ID" \
+          --set-key="StartupNotify" --set-value="true" \
+          --set-key="Categories" --set-value="Game;" \
+          --set-key="Keywords" --set-value="discord;overlay;" \
+          --set-key="StartupWMClass" --set-value="Overlayed" \
+          squashfs-root/usr/share/applications/overlayed.desktop
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:75]: why do you need to change so much of the .desktop file? is it so bad in the appimage? This should be fixed upstream is that's the case.

File: dev.overlayed.Overlayed.yaml [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+      - mkdir -p /app/{bin,share}
+      - cp -r squashfs-root/usr/share/icons /app/share/icons
+      - install -Dm755 squashfs-root/usr/bin/overlayed /app/bin/overlayed
+      - install -Dm644 squashfs-root/usr/share/applications/overlayed.desktop /app/share/applications/dev.overlayed.Overlayed.desktop
```
Comment (hfiguiere) [inline dev.overlayed.Overlayed.yaml:80]:
```suggestion
      - install -Dm755 squashfs-root/usr/bin/overlayed /app/bin/overlayed
      - install -Dm644 squashfs-root/usr/share/applications/overlayed.desktop /app/share/applications/dev.overlayed.Overlayed.desktop
      - cp -r squashfs-root/usr/share/icons /app/share/icons
```

File: dev.overlayed.Overlayed.yaml (hunk 13 above)
Comment (Hacksore) [reply dev.overlayed.Overlayed.yaml]: Was a mistake, I've fixed it to use `--socket=wayland` and `--socket=fallback-x11`.

File: dev.overlayed.Overlayed.yaml [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+      - mkdir -p /app/{bin,share}
+      - install -Dm755 squashfs-root/usr/bin/overlayed /app/bin/overlayed
+      # TODO: fix desktop file
+      # - install -Dm644 squashfs-root/usr/share/applications/overlayed.desktop /app/share/applications/dev.overlayed.Overlayed.desktop
+      # TODO: fix icons
+      # - cp -r squashfs-root/usr/share/icons /app/share/icons
+
+    sources:
+      - type: file
+        url: https://raw.githubusercontent.com/overlayeddev/overlayed/14a6d2c8b022f7d44de4a57350e8f30ff35ef7dc/distribution/dev.overlayed.Overlayed.metainfo.xml
```
Comment (bbhtt) [inline dev.overlayed.Overlayed.yaml:69]: This file is not being installed.

**Comment by Hacksore** [comment]
Update, the app currently will not start locally 😢.

So, I am going to close this and reopen a new one once I figure out the segfault blockers and get it running locally.

Repro with MVP of tauri v1 that attempts to convert appimage to flatpak for those interested.
https://github.com/Hacksore/flatpak-tauri
